src/bit_vec.rs

```diff
@@ -17,70 +17,63 @@
 use std::mem;
 
 use bitvec::{vec::BitVec, bits::Bits, cursor::Cursor, slice::BitSlice, boxed::BitBox};
-use byte_slice_cast::{AsByteSlice, ToByteSlice, FromByteSlice, Error as FromByteSliceError};
 
 use crate::codec::{Encode, Decode, Input, Output, Compact, Error};
 
-impl From<FromByteSliceError> for Error {
-	fn from(e: FromByteSliceError) -> Error {
-		match e {
-			FromByteSliceError::AlignmentMismatch {..} => "failed to cast from byte slice: alignment mismatch".into(),
-			FromByteSliceError::LengthMismatch {..} => "failed to cast from byte slice: length mismatch".into(),
-		}
-	}
-}
-
-impl<C: Cursor, T: Bits + ToByteSlice> Encode for BitSlice<C, T> {
+impl<C: Cursor, T: Bits + Encode> Encode for BitSlice<C, T> {
 	fn encode_to<W: Output>(&self, dest: &mut W) {
 		let len = self.len();
 		assert!(len <= u32::max_value() as usize, "Attempted to serialize a collection with too many elements.");
 		Compact(len as u32).encode_to(dest);
-		dest.write(self.as_slice().as_byte_slice());
+
+		for item in self.as_slice() {
+			item.encode_to(dest);
+		}
```

Comment:
I think we shouldn't iterate on element otherwise we loose optimization and result in encoding each byte on by one. Maybe call something like `self.as_slice().encode_to(dest)`

Reply:
This way additional length will be stored.